src/seismometer/controls/selection.py

```diff
@@ -112,59 +114,83 @@ def __init__(
             self.selection_widgets[key] = selection_widget
             selection_widget.observe(self._on_subselection_change, "value")
-        self.accordion = Accordion(
-            children=[
-                HBox(
-                    children=[self.selection_widgets[key] for key in self.selection_widgets],
-                    layout=Layout(width="max-content"),
-                )
-            ],
-            selected_index=0,
-        )
-        self.children = [self.accordion]
-        self.layout = Layout(width="max-content")
-        self.accordion.set_title(0, title)
-        self._on_subselection_change()
+        self.value_update_in_progress = False
+        self.title_box = HTML()
+        self.children = [
+            self.title_box,
+            Box(
+                children=[self.selection_widgets[key] for key in self.selection_widgets],
+                layout=Layout(
+                    display="flex",
+                    flex_flow="row wrap",
+                    align_items="flex-start",
+                    grid_gap="20px",
+                    border="solid 1px var(--jp-border-color1)" if border else None,
+                    padding="var(--jp-cell-padding)" if border else None,
+                ),
+            ),
+        ]
+        self.update_title_section(self.title)
+        self.observe(self._on_value_change, "value")
+        self._disabled = False
+
+    @property
+    def disabled(self) -> bool:
+        return self._disabled
+
+    @disabled.setter
+    def disabled(self, disabled: bool):
+        self._disabled = disabled
+        for widget in self.selection_widgets.values():
+            widget.disabled = disabled
 
     def _on_subselection_change(self, change=None):
         """Sets the observable value."""
-        self.value = {k: v for k, v in self.control_value.items() if v}
-
-    @property
-    def control_value(self) -> dict[str, tuple[str]]:
-        """Reads values from child widgets."""
-        return {key: self.selection_widgets[key].value for key in self.selection_widgets}
+        if self.value_update_in_progress:
+            return
+        if change and change["owner"] in self.selection_widgets.values():
+            self.value = {k: tuple(v.value) for k, v in self.selection_widgets.items() if len(v.value)}
 
-    @control_value.setter
-    def control_value(self, new_value):
-        """Updates the actual control value."""
-        for key in self.selection_widgets:
-            self.selection_widgets[key].control_value = new_value.get(key, ())
+    def _on_value_change(self, change=None):
+        """Bubble up changes."""
+        self.value_update_in_progress = True
+        updated_values = {key: self.value.get(key, ()) for key in self.selection_widgets}
+        for key, value in updated_values.items():
+            self.selection_widgets[key].value = self.value.get(key, ())
```

Comment:
replacing `self.value.get(key, ())` with `value` 

or

```for key in self.selection_widgets:
            self.selection_widgets[key].value = self.value.get(key, ()) ```

would be more efficient I believe

Reply:
But we also want to freeze the values of all the widgets when the request first comes in. As we loop over values, we end up changing them. 

It shouldn't matter since we end up skipping reverse updates in `_on_subselection_change` while this is happening, I wanted to make sure I only have an immutable object while I make my changes, rather than `self.value` which is mutable.